Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,334 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using UnityEngine;
+
+#if WINDOWS_UWP
+using Windows.Foundation;
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+using System.Threading.Tasks;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Enables text to speech using the Windows 10 <see cref="SpeechSynthesizer"/> class.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="SpeechSynthesizer"/> generates speech as a <see cref="SpeechSynthesisStream"/>. 
+    /// This class converts that stream into a Unity <see cref="AudioClip"/> and plays the clip using 
+    /// the <see cref="AudioSource"/> you supply in the inspector. This allows you to position the voice 
+    /// as desired in 3D space. One recommended approach is to place the AudioSource on an empty 
+    /// GameObject that is a child of Main Camera and position it approximately 0.6 units above the 
+    /// camera. This orientation will sound similar to Cortana's speech in the OS.
+    /// </remarks>
+    public class TextToSpeechManager : MonoBehaviour
+    {
+        // Inspector Variables
+        [Tooltip("The audio source where speech will be played.")]
+        public AudioSource audioSource;
+
+        // Member Variables
+        #if WINDOWS_UWP
+        private SpeechSynthesizer synthesizer;
+        #endif
+
+        // Static Helper Methods
+
+        /// <summary>
+        /// Converts two bytes to one float in the range -1 to 1 
+        /// </summary>
+        /// <param name="firstByte">
+        /// The first byte.
+        /// </param>
+        /// <param name="secondByte">
+        /// The second byte.
+        /// </param>
+        /// <returns>
+        /// The converted float.
+        /// </returns>
+        static private float BytesToFloat(byte firstByte, byte secondByte)
```

Comment:
BitConverter can't be used here for two reasons: The float value that Unity expects is little-endian and it must also be within the range of -1 to +1. This method does the conversion correctly for the expected audio floating point value and BitConverter would not.